scylla/tests/integration/session/schema_agreement.rs

```diff
@@ -0,0 +1,202 @@
+use std::collections::HashMap;
+use std::net::{IpAddr, SocketAddr};
+use std::sync::Arc;
+
+use assert_matches::assert_matches;
+use scylla::client::session::Session;
+use scylla::client::session_builder::SessionBuilder;
+use scylla::errors::{ExecutionError, RequestAttemptError, SchemaAgreementError};
+use scylla::policies::load_balancing::{NodeIdentifier, SingleTargetLoadBalancingPolicy};
+use scylla::response::query_result::QueryResult;
+use scylla::statement::Statement;
+use scylla_proxy::{
+    Condition, ProxyError, Reaction, RequestOpcode, RequestReaction, RequestRule, RunningProxy,
+    ShardAwareness, WorkerError,
+};
+use uuid::Uuid;
+
+use crate::utils::{setup_tracing, test_with_3_node_cluster, unique_keyspace_name};
+
+async fn run_some_ddl_with_paused_node(
+    coordinator: NodeIdentifier,
+    paused: usize,
+    session: &Session,
+    running_proxy: &mut RunningProxy,
+) -> Result<QueryResult, ExecutionError> {
+    // Prevent fetching schema version.
+    // This is similar to node being paused.
+    running_proxy.running_nodes[paused].change_request_rules(Some(vec![RequestRule(
+        Condition::and(
+            Condition::not(Condition::ConnectionRegisteredAnyEvent),
+            Condition::and(
+                Condition::RequestOpcode(RequestOpcode::Query),
+                Condition::BodyContainsCaseSensitive(Box::new(*b"system.local")),
+            ),
+        ),
+        // Simulates driver discovering that node is paused
+        RequestReaction::drop_connection(),
+    )]));
+
```

Comment:
:question: WDYM by node being _paused_? I guess processes stopped with SIGSTOP would have their TCP heartbeat still sent by the OS, so the connection wouldn't be dropped.
OTOH, what would happen upon SIGSTOP is lack of response to a keepalive. Assuming that our keepalive is an `OPTIONS -> SUPPORTED` roundtrip, we could simply drop `OPTIONS` frames.

Reply:
ok